src/main/java/com/google/cloud/security/scanner/actions/modifiers/JoinKnownGoodAndLiveStates.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.security.scanner.actions.modifiers;
+
+import com.google.cloud.dataflow.sdk.transforms.DoFn;
+import com.google.cloud.dataflow.sdk.values.KV;
+import com.google.cloud.dataflow.sdk.values.PCollectionView;
+import com.google.cloud.security.scanner.actions.modifiers.TagStateWithSource.StateSource;
+import com.google.cloud.security.scanner.primitives.GCPResource;
+import com.google.cloud.security.scanner.primitives.GCPResourceState;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Transform to join the known good and live state configurations.
+ */
+public class JoinKnownGoodAndLiveStates
```

Comment:
Java class namings should be noun-based (and methods are verbs).  So, it would be awesome if this class is like:
`KnownGoodAndLiveStatesJoiner`